### PR TITLE
Add CITATION.cff

### LINKED ISSUE #8760: Add CITATION.cff
### Current problem

I would like to cite pylint in a research article and it is unclear how pylint should be cited.

### Desired solution

I would like pylint to have a CITATION.cff file in the root of the repository.

[GitHub recommends](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files), and integrates support for, [CITATION.cff](https://citation-file-format.github.io/) files. Inclusion of this file would immediately have these benefits:

- Users will have the necessary metadata to properly cite pylint.
- GitHub can automatically generate both APA and BibTeX formatted citations. A "Cite this repository" option will automatically appear on the repo's sidebar when viewed on a desktop browser.
- Reference managers like Zotero will automatically detect the metadata so the user can save the citation to their library.

### Additional context

Examples of popular packages using this are [pandas](https://github.com/pandas-dev/pandas) and [tensorflow](https://github.com/tensorflow/tensorflow).

I would be happy to create an initial version of this and open a PR if there is interest in possibly merging it after refinement.

### PR DESCRIPTION
**Pull request by schackartk**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

This pull request adds a CITATION.cff file to the root of the repository which contains all the necessary metadata to properly cite Pylint.

Closes #8760

## Further Considerations

I have drafted an initial version of this file, but it is up to the maintainers what information to include. Some items for further consideration:

- Any preference on the author field? Some projects do not include any individuals, but I opted to include @Pierre-Sassoulas then wrap everyone else in "Pylint contributors". How this is worded is entirely optional.
- If including @Pierre-Sassoulas explicitly, do you have an ORCiD you would like to include?
- Any revisions of the abstract?
- CFF has a `release-date` field. Some projects put the year of first release there.
- CFF also has a `version` field, but it is generally not included. It must be kept up to date which adds friction, and most citation formats will not use the information anyway.

Also, I ran `tox -e docs`, but I am not sure if it ran succesfully. Certain files were changed in ways that I didn't think needed to be commited (`docs/user_guide/checkers/features.rst`, `docs/user_guide/configuration/all-options.rst`, and `doc/user_guide/messages/message_overview.rst`).

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you for opening the merge request !

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,16 @@
+cff-version: 1.2.0
+title: "Pylint: It's not just a linter that annoys you!"
+message: >-
+  If you use this software, please cite it using the metadata from this file.
+type: software
+authors:
+  - family-names: Sassoulas
+    given-names: Pierre
+  - name: "Pylint contributors"
```
[the remark this answers is not in the document]
Comment (schackartk) [reply CITATION.cff:9]: I will go with your second option. Should it be "Pylint" or "pylint" though?

**Comment by Pierre-Sassoulas** [comment]
> CFF has a release-date field. Some projects put the year of first release there.

That would be 2003

> CFF also has a version field, but it is generally not included. It must be kept up to date which adds friction, and most citation formats will not use the information anyway.

This can be updated automatically with tbump see https://github.com/pylint-dev/pylint/blob/main/tbump.toml#L22

> Also, I ran tox -e docs, but I am not sure if it ran succesfully. Certain files were changed in ways that I didn't think needed to be commited (docs/user_guide/checkers/features.rst, docs/user_guide/configuration/all-options.rst, and doc/user_guide/messages/message_overview.rst).

Yeah don't worry, this is not always up-to-date. Also there's a problem with the documentation check in the CI right now see #8762, so we're definitely going to update the doc in this one.

File: CITATION.cff (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pierre-Sassoulas) [reply CITATION.cff:9]: Not sure, ``pylint`` is used in package manager and the like so pylint without capitalization also make sense but I remember we talked about it at some point when working on the doc. Maybe another maintainer will remember what was decided.

**Comment by Pierre-Sassoulas** [comment]
Didn't think of the date that should change too, what about ``current`` ?

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,16 @@
+cff-version: 1.2.0
+title: "Pylint"
+message: >-
+  If you use this software, please cite it using the metadata from this file.
+type: software
+authors:
+  - family-names: Sassoulas
+    given-names: Pierre
```
Comment (Pierre-Sassoulas) [inline CITATION.cff:8]:
```suggestion
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Could you rebase on the upstream main branch please, @jacobtylerwalls fixed the issue with the doc check :)

File: CITATION.cff (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielNoord) [reply CITATION.cff:9]: We decided to leave it undecided I think haha
I think it makes sense to capitalise at the beginning of a sentence but keep it lower case anywhere else

**Comment by schackartk** [comment]
Yes, so sorry! I just wrapped up a different pull request and deleted the wrong repository >.< . I re-forked it and restored the branch, but I am unsure how to reopen the PR.

(Thanks for bearing with me, still a bit new to all this!)

**Comment by schackartk** [comment]
Reopen PR

**Comment by Pierre-Sassoulas** [comment]
I think you need to refork pylint ? 

repodeleted

**Comment by Pierre-Sassoulas** [comment]
Don't worry I re-created it in https://github.com/pylint-dev/pylint/pull/8777 and credited you by signing the commit with your email.

**Pull request closed without merging**